packages/runtime-core/src/vnode.ts

```diff
@@ -584,11 +584,16 @@ export function normalizeVNode(child: VNodeChild): VNode {
     return createVNode(Comment)
   } else if (isArray(child)) {
     // fragment
-    return createVNode(Fragment, null, child)
+    return createVNode(
+      Fragment,
+      null,
+      // #3666, avoid reference pollution when reusing vnode
+      child.map(c => c)
```

Comment:
There are many ways to clone an array, but `map` function maybe a bit slow. Here is the [bench](https://jsben.ch/OwuLW). IMO, we could use `slice`, `Array.from` or `spread`, but considered compatibility, maybe `slice` is better

Reply:
Sure, here I use `map` because it is left over from the original solution, check my first commit https://github.com/vuejs/vue-next/pull/3670/commits/b43c4c4dfd6c2ec28961842b4bc77bd01eb6cc4d#diff-9b2ba253038f7b71dbbbcb545e028d2d51d27d88b041b918c9eb102a83fef189R590